File: source/zombie/ai/permission/DefaultStatePermissions.java
// Decompiled on Sat Jan 17 08:24:00 EST 2026 with Zomboid Decompiler v0.2.3 using Vineflower.
package zombie.ai.permission;

import zombie.characters.IsoGameCharacter;

public class DefaultStatePermissions implements IStatePermissions {
    public static final DefaultStatePermissions Instance = new DefaultStatePermissions();

    @Override
    public boolean isDeferredMovementAllowed(IsoGameCharacter var1) {
        return true;
    }

    @Override
    public boolean isPlayerInputAllowed(IsoGameCharacter var1) {
        return true;
    }
}

File: ProjectZomboidSwitch/src/zombie/ai/permission/DefaultStatePermissions.cpp
#include "zombie/ai/permission/DefaultStatePermissions.h"

namespace zombie {
namespace ai {
namespace permission {
DefaultStatePermissions::DefaultStatePermissions() {}
DefaultStatePermissions::~DefaultStatePermissions() {}

DefaultStatePermissions& DefaultStatePermissions::instance() {
	static DefaultStatePermissions instance_;
	return instance_;
}

bool DefaultStatePermissions::isDeferredMovementAllowed(const IsoGameCharacter& character) const {
	// ...existing code...
}

bool DefaultStatePermissions::isPlayerInputAllowed(const IsoGameCharacter& character) const {
	// ...existing code...
}

bool DefaultStatePermissions::isPermissionAllowed(const IsoGameCharacter& character, const std::string& permissionName) const {
	if (permissionName == "deferredMovement") return isDeferredMovementAllowed(character);
	if (permissionName == "playerInput") return isPlayerInputAllowed(character);
	return false;
}

// Implementation now in header (inline)

} // namespace permission
} // namespace ai
} // namespace zombie
